### PR TITLE
Add option to decode STRING_EXT as int list

### PR DESCRIPTION
**Pull request by sharyari**
Erlang "strings" are in reality just lists of integers, where all the numbers are below 256, but In ETF-format, such integer lists get marked as STRING_EXT. Term currently has a setting ('byte_string') for choosing whether such lists should be decoded as bytes, or unicode strings.

This PR adds the option to decode them as int-lists, which is true to erlangs representation. This would be a third option for decode, i.e.
decode(data, options={'byte_string': repr}), where repr = 'str' | 'bytes' | 'int_list'

I also suggest that this be the default behaviour, as it is the only option where the data type is encoded deterministically (erlang list of ints always becomes a python list of ints, as opposed to the current behaviour, where we get either a list of ints or a string depending on the size of the integers in the list)

### CODE REVIEW

File: term/py_codec_impl.py [hunk 1]
```diff
@@ -147,6 +147,10 @@ def _create_atom_atom(name: bytes, encoding: str) -> Atom:
 
 def _get_create_str_fn(opt: str) -> Callable:
     """ A tool function to create either a str or bytes from a 8-bit string. """
+
+    def _create_int_list(name: bytes) -> bytes:
```
Comment (s2hc-johan) [inline term/py_codec_impl.py]: should be `-> list` right

Comment (sharyari) [reply term/py_codec_impl.py]: correct, I'll fix that

**Comment by sharyari** [comment]
> Also interesting, as we discussed, would be to check for the functionality of packing a list containing only integers with values < 256 as STRING_EXT.

I agree, but I regard that as purely an efficiency issue, which would preferably be adressed in another PR.

**Comment by kvakvs** [comment]
A list of integers in Unicode range is still a valid string, there is no easy way to know without the user of the library telling us. Unicode range is U+0-10FFFF

**Comment by s2hc-johan** [comment]
@kvakvs it's a valid string but it is in fact a list. I think it's this interpretation that is the issue in #2 as well. 

It will all come down to preferences. My vote is on making EXT_STRING a list when decoding.

**Comment by jonathansharyari** [comment]
One issue I have with this is that setting
`{'byte_string': 'int_list'}`
is really just gibberish. I propose the key is changed to 'string_ext', i.e.
`{'string_ext': 'int_list' | 'str' | 'bytes' }`

**Comment by s2hc-johan** [comment]
Well `'string_ext'` makes sense for decode. As we go forward we could try and come up with a name that makes sense and that would allow for settings that makes roundtrip work well

**Comment by s2hc-johan** [comment]
this was borked, merged #11 instead

**Pull request closed without merging**